### PR TITLE
Allow session cookies with environment variable

### PR DESCRIPTION
**Pull request by MeganerdDev**
# Closes: #4168
# What's Changed

- Adds "NAUTOBOT_SESSION_EXPIRE_AT_BROWSER_CLOSE" environment variable stored as Django value SESSION_EXPIRE_AT_BROWSER_CLOSE in settings.py
- Default value is False

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

# Notes
The only distinction between a session cookie and a persistent cookie is the 'Expires=' tag that specifies when a persistent cookie should expire. If a cookie has no 'Expires=' tag, then it is automatically interpreted as a session cookie

### CODE REVIEW

**Comment by MeganerdDev** [comment]
I have tested this to be working

**Comment by MeganerdDev** [comment]
I have changes from another task I was working on in this, stripping that out

**Comment by gsnider2195** [comment]
Can you add this to these files?

- `nautobot/core/templates/nautobot_config.py.j2`
- `nautobot/docs/user-guide/administration/configuration/optional-settings.md`

You can use `NAUTOBOT_CORS_ALLOW_ALL_ORIGINS` as an example of another django setting that we override

**Comment by MeganerdDev** [comment]
Documentation added, thanks for the assist @gsnider2195

File: nautobot/docs/configuration/optional-settings.md [hunk 1]
```diff
@@ -1141,6 +1141,19 @@ Please see the [official Django documentation on `MEDIA_ROOT`](https://docs.djan
...
+### SESSION_EXPIRE_AT_BROWSER_CLOSE
+
+Default: `False`
+
+Environment Variable: `NAUTOBOT_SESSION_EXPIRE_AT_BROWSER_CLOSE`
+
+Set to True to use session cookies instead of persistent cookies.
+Session cookies will expire when a browser is closed.
+
+Please see the [official Django documentation on `SESSION_EXPIRE_AT_BROWSER_CLOSE`](https://docs.djangoproject.com/en/4.2/ref/settings/#session-expire-at-browser-close) for more information.
```
Comment (glennmatthews) [inline nautobot/docs/configuration/optional-settings.md:1153]:
```suggestion
Please see the [official Django documentation on `SESSION_EXPIRE_AT_BROWSER_CLOSE`](https://docs.djangoproject.com/en/stable/ref/settings/#session-expire-at-browser-close) for more information.
```

File: nautobot/docs/configuration/optional-settings.md [hunk 2]
```diff
@@ -1149,6 +1162,8 @@ Environment Variable: `NAUTOBOT_SESSION_COOKIE_AGE`
 
 The age of session cookies, in seconds.
 
+Please see the [official Django documentation on `SESSION_COOKIE_AGE`](https://docs.djangoproject.com/en/4.2/ref/settings/#session-cookie-age) for more information.
```
Comment (glennmatthews) [inline nautobot/docs/configuration/optional-settings.md:1165]:
```suggestion
Please see the [official Django documentation on `SESSION_COOKIE_AGE`](https://docs.djangoproject.com/en/stable/ref/settings/#session-cookie-age) for more information.
```

**Review by glennmatthews — APPROVED** [review]
IIRC our usual pattern is to link to the "stable" Django docs rather than explicitly to a specific version of the docs. Otherwise, looks great!

File: nautobot/docs/configuration/optional-settings.md [hunk 3]
```diff
@@ -1141,6 +1141,19 @@ Please see the [official Django documentation on `MEDIA_ROOT`](https://docs.djan
...
 ---
 
+### SESSION_EXPIRE_AT_BROWSER_CLOSE
+
+Default: `False`
+
+Environment Variable: `NAUTOBOT_SESSION_EXPIRE_AT_BROWSER_CLOSE`
+
+Set to True to use session cookies instead of persistent cookies.
+Session cookies will expire when a browser is closed.
```
Comment (gsnider2195) [inline nautobot/docs/configuration/optional-settings.md:1151]:
```suggestion
If this is set to True, Nautobot will use browser-length cookies - cookies that expire as soon as the user closes their browser.

By default, `SESSION_EXPIRE_AT_BROWSER_CLOSE` is set to False, which means session cookies will be stored in users’ browsers for as long as [`SESSION_COOKIE_AGE`](#session_cookie_age).
```

**Review by gsnider2195 — APPROVED** [review]

File: changes/4169.added [hunk 4]
```diff
@@ -0,0 +1 @@
+Added environment variable NAUTOBOT_SESSION_EXPIRE_AT_BROWSER_CLOSE that allows for session cookies if set to True.
```
Comment (gsnider2195) [inline changes/4169.added:1]:
```suggestion
Added environment variable `NAUTOBOT_SESSION_EXPIRE_AT_BROWSER_CLOSE` to set the `SESSION_EXPIRE_AT_BROWSER_CLOSE` django setting which expires session cookies when the user closes their browser.
```

**Pull request merged**